File: Bucle while.js
// Parte 1: Bucle while para contar hasta 30
let contador = 0;
while (contador < 30) {
    contador++;
    console.log('Iteración', contador);
}

// Parte 2: Bucle while con if-else y manejo de errores
while (true) {
    try {
        // Solicitar entrada al usuario (en un entorno de navegador)
        let a = parseInt(prompt('Introduzca un valor mayor a 10: '));

        if (a > 10) {
            console.log('Es correcto');
            break; // Sale del bucle cuando la condición se cumple
        } else {
            console.log('Es incorrecto, pruebe de nuevo');
        }
    } catch (error) {
        console.log('Por favor, ingrese un número válido');
    }
}

File: Anyelo Carrillo C++/Bucle while.cpp
#include <iostream>
using namespace std;

int main() {
    // Primera parte: Bucle while para contar hasta 30
    int Contador = 0;
    while (Contador < 30) {
        Contador++;
        cout << "Iteración " << Contador << endl;
    }

    // Segunda parte: Bucle while con if-else y manejo de errores
    while (true) {
        int a;
        cout << "Introduzca un valor mayor a 10: ";
        cin >> a;

        if (cin.fail()) { // Verifica si la entrada no es un número
            cin.clear(); // Limpia el estado de error de cin
            cin.ignore(numeric_limits<streamsize>::max(), '\n'); // Ignora la entrada incorrecta
            cout << "Por favor, ingrese un número válido." << endl;
        } else {
            if (a > 10) {
                cout << "Es correcto" << endl;
                break; // Sale del bucle cuando la condición se cumple
            } else {
                cout << "Es incorrecto, pruebe de nuevo" << endl;
            }
        }
    }

    return 0;
}